test/e2e/pages/allocation-criteria.js

```diff
@@ -0,0 +1,62 @@
+import faker from 'faker'
+import { Selector } from 'testcafe'
+
+import { fillInForm } from '../_helpers'
+
+import Page from './page'
+
+class AllocationCriteriaPage extends Page {
+  constructor() {
+    super()
+    this.url = '/allocation/new/allocation-criteria'
+    this.nodes = {
```

Comment:
I think you may have misunderstood a previous comment. The name of this property now doesn't fit what it contains. This property does contain _fields_ so it would be better to use `this.fields = {}`. 

The other comment was when `this.fields` contained non-field elements of the page.

A clearer way to define this would probably be:

```javascript
this.fields = {
  prisonerCategory: Selector('#prisoner_category'),
  ...
}
this.errorMessages = [
  Selector('a').withAttribute('href','#prisoner_category'),
  ...
]
```

Reply:
Ok, I have separated the error bits.